api/src/api/users/user_routes.py

```diff
@@ -452,3 +455,52 @@ def user_update_saved_search(
     )
 
     return response.ApiResponse(message="Success")
+
+
+@user_blueprint.post("/<uuid:user_id>/api-keys")
+@user_blueprint.input(UserApiKeyCreateRequestSchema, location="json")
+@user_blueprint.output(UserApiKeyCreateResponseSchema)
+@user_blueprint.doc(responses=[200, 400, 401, 403])
+@user_blueprint.auth_required(api_jwt_auth)
+@flask_db.with_db_session()
+def user_create_api_key(
+    db_session: db.Session, user_id: UUID, json_data: dict
+) -> response.ApiResponse:
+    """Create a new API key for the authenticated user"""
+    logger.info("POST /v1/users/:user_id/api-keys")
```

Comment:
If you do this, every log for the rest of the request's lifecycle will have user_id automatically included.

```suggestion
) -> response.ApiResponse:
    """Create a new API key for the authenticated user"""
    add_extra_data_to_current_request_logs({"user_id": user_id})
    logger.info("POST /v1/users/:user_id/api-keys")
```